core/src/main/java/com/linecorp/armeria/server/docs/DocServiceFilter.java

```diff
@@ -0,0 +1,177 @@
+     */
+    static DocServiceFilter pluginName(String pluginName) {
+        requireNonNull(pluginName, "pluginName");
+        return (plugin, service, method) -> pluginName.equals(plugin);
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the service matches the
+     * specified {@code serviceName}.
+     */
+    static DocServiceFilter serviceName(String serviceName) {
+        requireNonNull(serviceName, "serviceName");
+        return (plugin, service, method) -> serviceName.equals(service);
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the plugin and service
+     * matches the specified {@code pluginName} and {@code serviceName}.
+     */
+    static DocServiceFilter serviceName(String pluginName, String serviceName) {
+        requireNonNull(pluginName, "pluginName");
+        requireNonNull(serviceName, "serviceName");
+        return (plugin, service, method) -> pluginName.equals(plugin) && serviceName.equals(service);
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the method matches the
+     * specified {@code methodName}.
+     */
+    static DocServiceFilter methodName(String methodName) {
+        requireNonNull(methodName, "methodName");
+        return (plugin, service, method) -> methodName.equals(method);
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the service and method
+     * matches the specified {@code serviceName} and {@code methodName}.
+     */
+    static DocServiceFilter methodName(String serviceName, String methodName) {
+        return (plugin, service, method) -> serviceName.equals(service) && methodName.equals(method);
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the plugin, service and
+     * method matches the specified {@code pluginName}, {@code serviceName} and {@code methodName}.
+     */
+    static DocServiceFilter methodName(String pluginName, String serviceName, String methodName) {
+        return (plugin, service, method) -> pluginName.equals(plugin) && serviceName.equals(service) &&
+                                            methodName.equals(method);
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the service matches the
+     * specified {@code regex}.
+     */
+    static DocServiceFilter servicePattern(String regex) {
+        requireNonNull(regex, "regex");
+        final Pattern pattern = Pattern.compile(regex);
+        return (plugin, service, method) -> pattern.matcher(service).matches();
+    }
+
+    /**
+     * Returns a {@link DocServiceFilter} which returns {@code true} when the name of the method matches the
+     * specified {@code regex}.
+     */
+    static DocServiceFilter methodPattern(String regex) {
```

Comment:
- I'm curious which would be better in this case, `Matcher.find()` or `matches()`?
- How about invoking `Matcher.matches()` with `plugin + ':' + service + '#' + method`? Then we would not need `servicePattern()` but only one factory method for a regex.

Reply:
I guess I have to use `find()`. Thank you!